lm_eval/evaluator.py

```diff
@@ -130,6 +134,17 @@ def simple_evaluate(
                 continue
 
         config = task_obj._config
+        if predict_only:
+            log_samples = True
+            eval_logger.info(
+                f"Processing {task_name} in output-only mode. Metrics will not be calculated!"
+            )
+            # we have to change the class properties post-hoc. This is pretty hacky.
+            task_obj._metric_fn_list = {"bypass": lambda x: None}
```

Comment:
maybe we can abstract this code that accesses private task class attributes into something like `ConfigurableTask.override_metrics()`? And/or an `ConfigurableTask.update_config()` method for updating the config it's based on